test/extensions/tracers/datadog/span_test.cc

```diff
@@ -0,0 +1,290 @@
+#include <datadog/clock.h>
+#include <datadog/collector.h>
+#include <datadog/expected.h>
+#include <datadog/id_generator.h>
+#include <datadog/logger.h>
+#include <datadog/sampling_priority.h>
+#include <datadog/span_data.h>
+#include <datadog/tags.h>
+#include <datadog/tracer.h>
+
+#include <cstdint>
+#include <datadog/json.hpp>
+#include <iomanip>
+#include <memory>
+#include <typeinfo>
+#include <utility>
+#include <vector>
+
+#include "source/common/tracing/null_span_impl.h"
+#include "source/extensions/tracers/datadog/span.h"
+#include "source/extensions/tracers/datadog/time_util.h"
+
+#include "test/mocks/tracing/mocks.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace datadog {
+namespace tracing {
+
+bool operator==(const TimePoint& left, const TimePoint& right) {
+  return left.wall == right.wall && left.tick == right.tick;
+}
+
+} // namespace tracing
+} // namespace datadog
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+namespace {
+
+class NullLogger : public datadog::tracing::Logger {
+public:
+  ~NullLogger() override = default;
+
+  void log_error(const LogFunc&) override {}
+  void log_startup(const LogFunc&) override {}
+
+  void log_error(const datadog::tracing::Error&) override {}
+  void log_error(datadog::tracing::StringView) override {}
+};
+
+struct MockCollector : public datadog::tracing::Collector {
+  datadog::tracing::Expected<void>
+  send(std::vector<std::unique_ptr<datadog::tracing::SpanData>>&& spans,
+       const std::shared_ptr<datadog::tracing::TraceSampler>&) override {
+    chunks.push_back(std::move(spans));
+    return {};
+  }
+
+  nlohmann::json config_json() const override {
+    return nlohmann::json::object({{"type", "Envoy::Extensions::Tracers::Datadog::MockCollector"}});
+  }
+
+  ~MockCollector() override = default;
+
+  std::vector<std::vector<std::unique_ptr<datadog::tracing::SpanData>>> chunks;
+};
+
+struct TestSetup {
```

Comment:
any reason not to make this a text fixture (make it a class derived from testing::Test, and use TEST_F macro below).

Also, independent of that, Envoy style is to suffix struct member vars with _ (overriding Google style).

Reply:
Will change the member names.

There was a voice in my head saying "you don't need `setUp` and `tearDown` methods -- a fixure obscure the simplicity in this case." Looking at it again, maybe this departure from common practice distracts more than simplifies.

Reply:
You would not define SetUp or TearDown -- there are empty default impls for those in testing::Test